src/main/java/org/opensearch/flowframework/function/ThrowingSupplier.java

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+package org.opensearch.flowframework.function;
+
+/**
+ * A supplier that can throw checked exception
+ *
+ * @param <T> method parameter type
+ * @param <E> Exception type
+ */
+@FunctionalInterface
+public interface ThrowingSupplier<T, E extends Exception> {
```

Comment:
The purpose of adding this class is to throw checked exceptions in static initialization blocks where Java doesn't allow to do it, enum constructors in this case. I prefer to not use `Callable` because it introduces an exception when we use `Callable.call()`. Therefore, in order to capture any exception thrown by `mappingCallable.call()`, we need to assign a default value to the `mapping`, like `null`, or logging it. I think it's more complicated to handle. But using this `ThrowingSupplier`, we don't need to care about all this. I see AD is using the same method for their Index Enum class. ML-Commons is not using any wrapper because they don't load index mapping from a json file, so they don't need worry about handle exceptions in an enum class

Reply:
Right, all the wrapping handles the use in static initialization (see later comments with wrapped callable suggestion).  Both work, it's probably more of a usage/style question where a "supplier" tends to be retrieving a value that exists somewhere close/fast while a "callable" needs to call out somewhere else to fetch a value and may be slower, and I feel that was the case here.